main.cpp

```diff
@@ -117,17 +132,25 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
-    
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
-    
-    U <#name3#>;
+    T t1(1, "first_T");                                             //6
+    T t2(2, "second_T");                                             //6
+
+    Comparator f;                                            //7
+    auto *smaller = f.compare(&t1, &t2);                              //8
```

Comment:
put the `*` next to the type name, not the variable name.
it makes it more clear that it's a pointer.   This also aligns with the JUCE Coding Standard
- [ ] revise